[G2M] data key exploration and chart-system (plotly line chart) stories

woozyking opened the pull request:
![line charts](https://user-images.githubusercontent.com/2837532/158270770-cc271d39-248a-480f-91a1-63d5b29e6762.png)

woozyking committed:
useSample - detect possible data keys and parsed typedKyes (by value types)

woozyking committed:
story - demo with @eqworks/chart-system highlighting data keys exploration

woozyking committed:
story - factor out common components for various stories

woozyking committed:
useExplorer - add API fetching loading and error states

vxsl commented:
nice :+1: 

the `Explore:` text input loses focus everytime I type a character because it is momentarily replaced with `Loading...` and then re-rendered. Makes it difficult to change the input.

woozyking commented:
> the Explore: text input loses focus everytime I type a character because it is momentarily replaced with Loading... and then re-rendered. Makes it difficult to change the input.

oh ye that was with the vanilla input. #3 uses lumen labs components so should be a bit more intuitive

hyx131 commented on src/index.js:
```diff
@@ -43,21 +56,49 @@ export const useSample = (sample) => {
+  // set parsed keys by inferred value type, react to keys or data change
+  const [typedKeys, setTypedKeys] = useState({})
+  useEffect(() => {
+    if (keys && keys.length > 0 && data && data.length > 0) {
```
or `if (keys?.length && data?.length)` 🌚

woozyking replied:
revised in [c789f4f](https://github.com/EQWorks/api-explorer/pull/2/commits/c789f4fc2317940e14e22184e5d53e55402be738) along some other logic simplification/cleanup

hyx131 reviewed (approved):
looks good!

woozyking committed:
useSample - simplify path and key seeking logic with optional chaining and implied truthy

- based on [review comment pull/2#discussion_r829138009](https://github.com/EQWorks/api-explorer/pull/2#discussion_r829138009)
